mart/attack/composer.py

```diff
@@ -7,29 +7,36 @@
 from __future__ import annotations
 
 import abc
-from typing import Any
+from typing import Any, Iterable
 
 import torch
 
 
 class Composer(abc.ABC):
     def __call__(
         self,
-        perturbation: torch.Tensor | tuple,
+        perturbation: torch.Tensor | Iterable[torch.Tensor],
         *,
-        input: torch.Tensor | tuple,
-        target: torch.Tensor | dict[str, Any] | tuple,
+        input: torch.Tensor | Iterable[torch.Tensor],
+        target: torch.Tensor | Iterable[torch.Tensor | dict[str, Any]],
```

Comment:
Do you mean 
```python
target: torch.Tensor | Iterable[torch.Tensor] | Iterable[dict[str, Any]]
```
?

Reply:
I think that's the same thing?

Reply:
I don't think so. Here's an example from `mypy`
```python
x: list[int | str] = [3, 5, "test", "fun"]
```
https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html

As we assume `Iterable` only has one type of elements, the annotation ought to be `Iterable[torch.Tensor] | Iterable[dict[str, Any]]`?

Reply:
Fixed in [10106df](https://github.com/IntelLabs/MART/pull/134/commits/10106df7bb216c45001f38df090100540688df4e).